### PR TITLE
test: Add commit message checks

### PR DESCRIPTION
**Pull request by justinprieto**
Adds GitHub Action to check commit message format

### CODE REVIEW

**Review by pjones-cedar — APPROVED** [review]
- [x] I fully understand what the code is doing and its context
- [x] The code is the best way to accomplish this problem (I would code it similarly)
- [x] The code is well organized, easy to understand and matches Cedar design patterns and best practices
- [ ] The code handles edge cases (that I can think of) and has tests to validate them
- [ ] The appropriate people are reviewing (knowledge and security)
- [ ] I tested the code personally (Not required)

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -9,16 +9,30 @@ jobs:
...
+        with:
+          fetch-depth: 0
+      - name: Set up Node
+        uses: actions/setup-node@v3
+      - name: Check commit syntax
+        # Install v16.2.3 (by commit ID)
+        run: |
+          yarn add @commitlint/cli#9128c3d5c9c9fedfb9969cc9ba4d38bdd883642f @commitlint/config-conventional#9128c3d5c9c9fedfb9969cc9ba4d38bdd883642f
+          echo "module.exports = {extends: ['@commitlint/config-conventional']}" > commitlint.config.js
+          yarn run commitlint --from HEAD~${{ github.event.pull_request.commits }} --to HEAD
```
Comment (juanrgon) [inline .github/workflows/tests.yml:38]: Are there docs that you followed for this?

Comment (justinprieto) [reply .github/workflows/tests.yml:38]: Adapated from "getting started" section [here](https://www.npmjs.com/package/@commitlint/config-conventional)

Comment (juanrgon) [reply .github/workflows/tests.yml:38]: Probably should include a comment with a link to that section

Comment (justinprieto) [reply .github/workflows/tests.yml:39]: Done

**Review by pjones-cedar — APPROVED** [review]
- [ ] I fully understand what the code is doing and its context
- [ ] The code is the best way to accomplish this problem (I would code it similarly)
- [ ] The code is well organized, easy to understand and matches Cedar design patterns and best practices
- [ ] The code handles edge cases (that I can think of) and has tests to validate them
- [ ] The appropriate people are reviewing (knowledge and security)
- [ ] I tested the code personally (Not required)

**Review by juanrgon — APPROVED** [review]
- [x] I fully understand what the code is doing and its context
- [x] The code is the best way to accomplish this problem (I would code it similarly)
- [x] The code is well organized, easy to understand and matches Cedar design patterns and best practices
- [x] The code handles edge cases (that I can think of) and has tests to validate them
- [x] The appropriate people are reviewing (knowledge and security)
- [ ] I tested the code personally (Not required)
